src/user_lists/serializers.py

```diff
@@ -1,20 +1,157 @@
+from django.core.exceptions import ObjectDoesNotExist
 from rest_framework import serializers
-from utils.serializers import DefaultAuthenticatedSerializer
 
-from .models import List
+from feed.models import FeedEntry
+from feed.serializers import FeedEntrySerializer, serialize_feed_metrics
+from paper.models import Paper
+from researchhub_document.related_models.constants.document_type import PAPER
+from researchhub_document.related_models.researchhub_post_model import ResearchhubPost
 
+from .models import List, ListItem
+
+
+ 
+READ_ONLY_FIELDS = ["id", "created_date", "updated_date", "created_by", "updated_by"]
+
+LIST_WRITABLE_FIELDS = ["is_public", "name"]
+LIST_ITEM_WRITABLE_FIELDS = ["is_public", "parent_list", "unified_document"]
+
+LIST_BASE_FIELDS = READ_ONLY_FIELDS + LIST_WRITABLE_FIELDS
+LIST_ITEM_FIELDS = READ_ONLY_FIELDS + LIST_ITEM_WRITABLE_FIELDS
+
+
+class ListSerializer(serializers.ModelSerializer):
+    items_count = serializers.SerializerMethodField()
+
+    class Meta:
+        model = List
+        fields = LIST_BASE_FIELDS + ["items_count"]
+        read_only_fields = READ_ONLY_FIELDS
+
+    def get_items_count(self, obj):
+        return getattr(obj, "items_count", 0)
+
+
+class ListItemReadSerializer(serializers.ModelSerializer):
+    unified_document = serializers.SerializerMethodField(method_name="get_feed_entry")
```

Comment:
This part is confusing because the metho_name is `get_feed_entry` but we are assigning to `unified_document`. The two are different structures. sounds like we need to change to `feed_entry`?